source/agora/consensus/EnrollmentManager.d

```diff
@@ -811,6 +812,28 @@ public class EnrollmentManager
         catch (Exception ex)
         {}
     }
+
+    /***************************************************************************
+
+        Gets the number of a valid validators at a next block height.
+
+        Params:
+            block_height = the next block height
+            enrolls = the enrollment data included when creating blocks
+
+    ***************************************************************************/
+
+    public ulong getAvailableCount (ulong block_height, ref Enrollment[] enrolls) @safe nothrow
```

Comment:
`getValidatorCount` would be a better name, and it should just forward to the implementation in `ValidatorSet`.

You don't need to pass the enrollments here. Think about it, if a block will contain enrollments it means the validator set will not be empty.

There is only a single protocol check that we need to do:

- If the enrollments in the proposed `ConsensusData` are empty, then check the ValidatorSet and make sure that there will still be valid enrollments for the proposed new block height.

Reply:
Shouldn't there be more than X validators?
If only one validator is needed, I agree with you.
However, When we need several, shouldn't we remove the same one and check the number?

Reply:
What I mean is, is that the calling function can already do this. It doesn't need to pass the `Enrollment[]` here. It could either pass the length, or it could calculate it at the call site.

So the code in Nominator could do something like this:

```D
if (consensus_data.enrollments.length + enrollment_man.getValidatorCount(proposed_block_height) < MIN_VALIDATOR_COUNT)
    return false;
```

Reply:
I understood your suggestion accurately. Thank you very much for your good opinion.

Reply:
I realized late that `ConsensusData.enrollments` and `ValidatorSet` cannot have the same data. @AndrejMitrovic , I'm sorry. It's all fixed.